### DISCUSSION TITLE
Default Values not working in MUI Checkbox

### DISCUSSION THREAD

**Discussion by azad-steadybiz** [react-hook-form/react-hook-form#2589] [Q&A]
I am saving MUI checkbox value in a array of string but default values not showing the checkbox as checked. But When I am submitting the form, I am getting the defaultValues.

Same thing working fine with type="checkbox"

CSB: https://codesandbox.io/s/dazzling-cray-en573?file=/src/App.js

**Comment by azad-steadybiz** [comment]
@bluebill1049

**Comment by azad-steadybiz** [comment]
@bluebill1049  Please reply.

**Comment by torian257x** [comment]
maybe need to wrap in controller component of rhf https://react-hook-form.com/api#Controller

**Comment by bluebill1049** [comment]
<img width="979" alt="Screen Shot 2020-08-14 at 8 49 02 am" src="https://user-images.githubusercontent.com/10513364/90194595-0221d680-de0b-11ea-8c77-32faf9f86cd6.png">
Take a look at the doc for Controller: there is a codesandbox there.

**Reply by azad-steadybiz** [reply]
If checkbox has exposed ref attributed then why to use controller?

**Reply by bluebill1049** [reply]
good question, because MUI have an animation which controlled by the state.

**Reply by azad-steadybiz** [reply]
It was working fine without defaultValues, So I need to wrap the checkbox with controller this is the only solution?

**Reply by bluebill1049** [reply]
yes, use Controller is safer.

**Reply by azad-steadybiz** [reply]
Still not working with defaultValues, 
Does I need to do something like this https://codesandbox.io/s/material-demo-639rq?file=/demo.js

but if we use onChange then what the purpose of RHF?

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by azad-steadybiz** [reply]
That "official" codesandbox example doesn't contain the case with checkbox array with default values though

**Reply by bluebill1049** [reply]
take a look at this issue: https://github.com/react-hook-form/react-hook-form/issues/1517

**Reply by azad-steadybiz** [reply]
Thanks
